### ISSUE TITLE
Localkube binary not found in VirtualBox

### ISSUE THREAD

**Issue by markus1189** [kubernetes/minikube#2307]
**This is a BUG REPORT**

**Environment**:
```
minikube version: v0.23.0                                                                                                                                                                                          
                                                                                                                                                                                                                   
OS:                                                                                                                                                                                                                
NAME=NixOS                                                                                                                                                                                                         
ID=nixos                                                                                                                                                                                                           
VERSION="18.03pre121925.3eccd0b11d1 (Impala)"                                                                                                                                                                      
VERSION_CODENAME=impala                                                                                                                                                                                            
VERSION_ID="18.03pre121925.3eccd0b11d1"                                                                                                                                                                            
PRETTY_NAME="NixOS 18.03pre121925.3eccd0b11d1 (Impala)"                                                                                                                                                            
HOME_URL="https://nixos.org/"                                                                                                                                                                                      
SUPPORT_URL="https://nixos.org/nixos/support.html"                                                                                                                                                                 
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"                                                                                                                                                           
                                                                                                                                                                                                                   
VM driver:                                                                                                                                                                                                         
    "DriverName": "virtualbox",                                                                                                                                                                                    
                                                                                                                                                                                                                   
ISO version                                                                                                                                                                                                        
        "Boot2DockerURL": "file://~/.minikube/cache/iso/minikube-v0.23.6.iso"
```

**What happened**:  

Running `helm init` result in:
```
$ helm init
Error: error installing: Post https://192.168.99.100:8443/apis/extensions/v1beta1/namespaces/kube-system/deployments: dial tcp 192.168.99.100:8443: getsockopt: connection refused
```

When doing `minikube logs` I saw the following:

```
Dec 12 07:51:00 minikube systemd[1]: localkube.service: Service hold-off time over, scheduling restart.                                                                                                            
Dec 12 07:51:00 minikube systemd[1]: Stopped Localkube.                                                                                                                                                             Dec 12 07:51:00 minikube systemd[1]: Starting Localkube...                                                                                                                    
Dec 12 07:51:00 minikube systemd[1]: localkube.service: Main process exited, code=exited, status=203/EXEC                                                                                                           Dec 12 07:51:00 minikube systemd[1]: Failed to start Localkube.                                                                                                                                      
Dec 12 07:51:00 minikube systemd[1]: localkube.service: Unit entered failed state.                                                                                                                                  Dec 12 07:51:00 minikube systemd[1]: localkube.service: Failed with result 'exit-code'.    
```

So I did a `minikube ssh` and inside there a `journalctl -u localkube.service`:

```
Dec 12 07:51:00 minikube systemd[1]: Starting Localkube...                                                                                                                                                         
Dec 12 07:51:00 minikube systemd[1]: localkube.service: Main process exited, code=exited, status=203/EXEC                                                                                                          
Dec 12 07:51:00 minikube systemd[1]: Failed to start Localkube.                                                                                                                                                    
Dec 12 07:51:00 minikube systemd[1]: localkube.service: Unit entered failed state.                                                                                                                                 
Dec 12 07:51:00 minikube systemd[1]: localkube.service: Failed with result 'exit-code'.                                                                                                                            
Dec 12 07:51:03 minikube systemd[1]: localkube.service: Service hold-off time over, scheduling restart. 
```

Spending some more time I found the following line:
```
Dec 12 07:46:50 minikube systemd[5454]: localkube.service: Failed at step EXEC spawning /usr/local/bin/localkube: No such file or directory                                                                        
```

Turn out that there is no `/usr/local/bin/localkube`.  I found it instead at `/localkube`

**What you expected to happen**:

Finding localkube at `/usr/local/bin/localkube` instead of at `/localkube`.

**How to reproduce it** (as minimally and precisely as possible):

Do `minikube start` and notice it failing.

**Temporary Fix**: For now I just manually did a `minikube ssh` and `cp`ed the `localkube` binary into the correct directory.

**Comment by r2d4** [comment]
I don't know why localkube wouldn't be in the right directory.  Can you run `minikube start --v 10 --logtostderr`?

**Comment by markus1189** [comment]
Here is the full output: [minikube.log](https://github.com/kubernetes/minikube/files/1554387/minikube.log)

I can see for example:
```
I1213 07:44:39.551784   29706 ssh_runner.go:57] Run: sudo rm -f /localkube                                                                                                                                         
```

but it's still there:

```
$ minikube ssh
                         _             _
            _         _ ( )           ( )
  ___ ___  (_)  ___  (_)| |/')  _   _ | |_      __
/' _ ` _ `\| |/' _ `\| || , <  ( ) ( )| '_`\  /'__`\
| ( ) ( ) || || ( ) || || |\`\ | (_) || |_) )(  ___/
(_) (_) (_)(_)(_) (_)(_)(_) (_)`\___/'(_,__/'`\____)

$ ls -hl /localkube
-rwxr-xr-x 1 root root 149M Dec 13 06:44 /localkube
$ ls -hl /usr/local/bin/
total 0
lrwxrwxrwx 1 root root 21 Oct 19 17:20 vmhgfs-fuse -> ../../bin/vmhgfs-fuse
```

**Comment by markus1189** [comment]
Nevermind, it's a problem in NixOS and how the minikube is being packaged.  Should be fixed by NixOS/nixpkgs/#32625

**Comment by r2d4** [comment]
Ah I've seen that before.  I'm not super familiar with nix pkgs, but I think that setup can be greatly simplified now that we don't depend on localkube as much.

**Comment by markus1189** [comment]
I would be very interested in that, can you give a pointer to what changed exactly?

**Comment by r2d4** [comment]
We used to package localkube with the minikube binary, through go-bindata.  Minikube would then extract that binary into the VM.  

Since we are soon deprecating localkube, we removed this bundling, and minikube just fetches localkube from our GCS bucket.  

The nix-pkg looks like its building localkube from source, since we weren't releasing it with the kind of granularity that someone needed.  Since we won't be upgrading localkube nearly as often, and deferring most of the cluster setup to kubeadm, this won't be needed in the future.

So the nix-pkg should just download the minikube binary and remove the custom building and bundling of localkube.
